internal/validation/validation_test.go

```diff
@@ -40,31 +53,47 @@ func TestValidate(t *testing.T) {
 
 	schemas := make([]*ast.Schema, len(testData.Schemas))
 	for i, schemaStr := range testData.Schemas {
-		schemas[i] = schema.New()
-		err := schema.Parse(schemas[i], schemaStr, false)
+		s := schema.New()
+
+		s.Directives["oneOf"] = &ast.DirectiveDefinition{
+			// graphql-js includes support for @oneOf, currently in RFC
+			// This is not available in graphql-go, nor is it expected to be unless the RFC is accepted
+			// See https://github.com/graphql/graphql-js/pull/3513 & https://github.com/graphql/graphql-spec/pull/825/
+			Name:      "oneOf",
+			Desc:      "Indicates exactly one field must be supplied and this field must not be `null`.",
+			Locations: []string{"INPUT_OBJECT"},
+		}
```

Comment:
This is just a validation, IMO, and we can easily include that when I make the `query` package public and allow a middleware with access to AST before actually executing the query. This way every user can validate whatever they want without including this in the library. This can be an example added to the tests/docs.

Reply:
I'd leave it here since it validates that we can add this directive here but it doesn't enforce any behavior.